Mathlib/Analysis/SpecialFunctions/Log/NegMulLog.lean

```diff
@@ -52,18 +52,133 @@ lemma hasDerivAt_mul_log {x : ℝ} (hx : x ≠ 0) : HasDerivAt (fun x ↦ x * lo
+    linarith
+    exact abs_eq
+  · have h := log_lt_log hx h
+    simp only [log_exp] at h
+    rw [abs_of_pos (by linarith)]
+    linarith
+
+lemma not_eventually_bounded_derivative_zero_mul_log (D : ℝ) :
+    ¬ ∀ᶠ (x : ℝ) in 𝓝 0, |x * x.log - x * D| ≤ |x| := by
+  simp [eventually_nhdsWithin_iff, Metric.eventually_nhds_iff]
+  intro x hx
+  exists min (2⁻¹ * x) (exp (D - 2))
+  have xhalf_gt0 : 0 < 2⁻¹ * x := by norm_num [hx]
+  constructor
+  · exact abs_min_lt_of_nonneg (le_of_lt xhalf_gt0) (exp_nonneg (D - 2)) (by norm_num; linarith)
+  · have exists_pos : 0 < |min (2⁻¹ * x) (rexp (D - 2))| := by
+      simp only [one_div, abs_pos]
+      exact ne_of_gt (lt_min xhalf_gt0 (exp_pos (D - 2)))
+    simp only [← mul_sub, one_div, abs_mul]
+    apply (lt_mul_iff_one_lt_right exists_pos).mpr
+    by_cases min_l : (2⁻¹ * x) < (rexp (D - 2))
+    · rw [min_eq_left_of_lt min_l]
+      exact NegMulLog.one_lt_log_sub_const_of_lt_exp_sub (2⁻¹ * x) D xhalf_gt0 min_l
+    · simp only [min_eq_right (le_of_not_lt min_l), log_exp, sub_sub_cancel_left, abs_neg,
+        Nat.abs_ofNat, Nat.one_lt_ofNat]
+
+-- TODO inline or refactor or find home
+/- In the hypothesis, D stands for a hypothetical derivative `deriv f x0`. -/
+lemma not_DifferentiableAt_of_not_eventuallly_bounded_derivative (f : ℝ → ℝ) (x0 : ℝ)
+    (hf : ∀ D, ¬ ∀ᶠ (x : ℝ) in 𝓝 0, |f (x0 + x) - f x0 - x * D| ≤ |x|) :
+    ¬ DifferentiableAt ℝ f x0 := by
+  intro h
+  have := hasDerivAt_iff_isLittleO_nhds_zero.mp (DifferentiableAt.hasDerivAt h)
+  simp only [zero_add, log_zero, mul_zero, sub_zero, smul_eq_mul] at this
+  have := Asymptotics.IsLittleO.bound this
+  simp only [norm_mul, norm_eq_abs] at this
+  have := @this 1 (by norm_num)
+  simp only [one_mul] at this
+  have := hf (deriv f x0)
+  contradiction
+
+lemma not_DifferentiableAt_log_mul_zero :
+    ¬ DifferentiableAt ℝ (fun x ↦ x * log x) 0 := by
+  apply not_DifferentiableAt_of_not_eventuallly_bounded_derivative (fun x ↦ x * log x) 0
+  simp only [zero_add, log_zero, mul_zero, sub_zero]
+  exact not_eventually_bounded_derivative_zero_mul_log
+
+/-- Not differentiable, hence `deriv` has junk value zero. -/
+lemma deriv_mul_log_zero : deriv (fun x ↦ x * log x) 0 = 0 :=
+  deriv_zero_of_not_differentiableAt not_DifferentiableAt_log_mul_zero
+
+open Filter in
+lemma tendsto_deriv_mul_log_nhdsWithin_zero :
+    Tendsto (deriv (fun x ↦ x * log x)) (𝓝[>] 0) atBot := by
+  have : (deriv (fun x ↦ x * log x)) =ᶠ[𝓝[>] 0] (fun x ↦ log x + 1) := by
+    apply eventuallyEq_nhdsWithin_of_eqOn
+    intro x hx
+    rw [deriv_mul_log]
+    simp only [Set.mem_Ioi, ne_eq]
+    exact ne_of_gt hx
+  simp only [tendsto_congr' this, tendsto_atBot_add_const_right, tendsto_log_nhdsWithin_zero_right]
+
+section NotContinuousAtOfTendstoNhdsWithin
+-- TODO put elsewhere, generalize, maybe add other nhdsWithin variants?
+
+open Filter
+
+lemma not_continuousAt_of_tendsto_nhdsWithin_Ioi_atTop {f : ℝ → ℝ} {x : ℝ}
+    (hf : Tendsto f (𝓝[>] x) atTop) :
```

Comment:
I think these could be generalized to something like
```
lemma not_continuousAt_of_tendsto {f : ℝ → ℝ} {l₁ l₂ : Filter ℝ} (x : ℝ)
    (hf : Tendsto f l₁ l₂) [l₁.NeBot] (hl₁ : l₁ ≤ 𝓝 x) (hl₂ : l₂ ⊓ 𝓝 (f x) = ⊥) :
    ¬ ContinuousAt f x := by sorry
```
(but properly generalized to more than just real functions). Or they could just be inlined since the proofs are quite short anyway. I think having the lemma would be nice, since it's a bit more discoverable than having to resort to lemmas about tendsto.

Reply:
Assuming I could prove that, not sure how to use this `not_continuousAt_of_tendsto`. For example, to specialize to my case, automation doesn't seem to do it and the best I could come up with is
```lean4
lemma not_continuousAt_of_tendsto_nhdsWithin_Ioi_atTop {f : ℝ → ℝ} {x : ℝ}
    (hf : Tendsto f (𝓝[>] x) atTop) :
    ¬ ContinuousAt f x := by
  apply not_continuousAt_of_tendsto x hf nhdsWithin_le_nhds
  apply inf_eq_bot_iff.mpr
  use Set.Ioi (f x + 2)
  constructor
  · exact Ioi_mem_atTop (f x + 2)
  · use Set.Ioo (f x - 1) (f x + 1)
    constructor
    apply Ioo_mem_nhds (by linarith) (by linarith)
    rw [Set.Ioi_inter_Ioo]
    simp only [max_lt_iff, add_lt_add_iff_left, Nat.not_ofNat_lt_one, false_and,
      not_false_eq_true, Set.Ioo_eq_empty]
```

Reply:
As requested by Yael, this is now in a new PR https://github.com/leanprover-community/mathlib4/pull/14594